apps/web/src/app/(signing)/sign/[token]/form.tsx

```diff
@@ -56,19 +58,29 @@ export const SigningForm = ({
   // Keep the loading state going if successful since the redirect may take some time.
   const isSubmitting = formState.isSubmitting || formState.isSubmitSuccessful;
 
+  const fieldsRequiringValidation = useMemo(
+    () =>
+      fields.filter((field) => {
+        const isRequired = field.fieldMeta && ZFieldMetaSchema.parse(field.fieldMeta)?.required;
+
+        return isRequired || !isAdvancedField(field.type) ? !field.inserted : false;
```

Comment:
can we simplify this statement. I'd rather have it over multiple lines than the current `||` into a ternary 😄